src/components/BaseLayerToggle.tsx

```diff
@@ -62,32 +78,134 @@ const InnerBackground = styled.span<InnerImageProps>`
   `};
 
   @media (-webkit-min-device-pixel-ratio: 2), (min-resolution: 192dpi) {
-    background: ${({
-      type,
-    }) => `url(${BASE_LAYER_STYLE[type].background[1]}) no-repeat top left /
+    background: ${({ type }) => `url(${
+      BASE_LAYER_STYLE[type as BaseLayerType].background[1]
+    }) no-repeat top left /
       120px 44px`};
   }
 `
+const Menu = styled(ContextMenu)`
+  & > button {
+    height: 44px;
+    min-width: 44px;
+    justify-content: center;
+    cursor: pointer;
+  }
+  div[role='menu'] {
+    justify-content: flex-start;
+    white-space: nowrap;
+  }
+`
 
-const InnerSpan = styled.span<InnerSpanProps>`
-  padding: ${themeSpacing(3, 4)};
-  display: inline;
-
-  ${({ open, theme }) => {
-    return open
-      ? `
-          background-color: ${themeColor('secondary')({ theme })};
-          color: ${themeColor('tint', 'level1')({ theme })};
-        `
-      : ``
-  }}
+const CheckmarkIcon = styled(Icon)`
+  padding-left: ${themeSpacing(3)};
 `
 
-const BaseLayerToggle: React.FC<Props> = ({ open, type, onClick }) => (
-  <StyledButton variant="blank" onClick={() => onClick && onClick(!open)}>
-    <InnerBackground type={type}>{type?.toString()}</InnerBackground>
-    <InnerSpan open={open}>&hellip;</InnerSpan>
-  </StyledButton>
-)
+const StyledContextMenuItem = styled(ContextMenuItem)`
+  justify-content: flex-start;
+  padding-left: ${themeSpacing(6)};
+  & > span:first-of-type {
+    padding-left: 0;
+    margin: ${themeSpacing(1, 1, 1, -4)};
+  }
+`
+
+const BaseLayerToggle: React.FC = () => {
+  const [toggleBaseLayerType, setToggleBaseLayerType] = useState(
+    BaseLayerType.Topo,
+  )
+  const [selectedLayer, setSelectedLayer] = useState({
+    [BaseLayerType.Aerial]: AERIAL_AMSTERDAM_LAYERS[0].urlTemplate,
+    [BaseLayerType.Topo]: DEFAULT_AMSTERDAM_LAYERS[0].urlTemplate,
+  })
+
+  const handleToggle = () => {
+    setToggleBaseLayerType(
+      toggleBaseLayerType === BaseLayerType.Aerial
+        ? BaseLayerType.Topo
+        : BaseLayerType.Aerial,
+    )
+  }
+
+  const handleChange = (
+    e: React.FormEvent<HTMLSelectElement>,
+    layers: MapLayer[],
+  ) => {
+    const { value } = e.currentTarget
+
+    const layer = layers.find(({ id }) => id === value)
+
+    if (layer) {
+      setSelectedLayer({
+        ...selectedLayer,
+        [toggleBaseLayerType]: layer.urlTemplate,
+      })
+    }
+  }
+
+  return (
+    <StyledButton>
+      {/*
+      // @ts-ignore */}
+      <ToggleButton
+        variant="blank"
+        onClick={handleToggle}
+        type={toggleBaseLayerType}
```

Comment:
This statement seems to cause problems as `type` is used internally somehere. If I change this to `layerType` all type errors seem to go away.

Reply:
nice one!